ci/Jenkinsfile-scan

```diff
@@ -14,135 +14,221 @@
-                            replacementPattern: '',
-                            scanExcludesPattern: '',
-                            scanIncludesPattern: '',
-                            scanName: "master-${currentBuild.number}-${currentBuild.startTimeInMillis}",
-                            uploadExcludesPattern: '**/.cache/**/*.jar,**/*dep*/**/*.jar,**/lib/**/*.jar,**/*repo*/**/*.jar,**/it/**/*.jar,**/sample*.jar',
-                            uploadIncludesPattern: '**/*.jar',
-                            waitForScan: true,
-                            vid: "$VERACODE_ID",
-                            vkey: "$VERACODE_KEY"
-                    }
-                }
-            }
+      }
+    }
+    stage("Veracode SCA (Source Clear Analysis)") {
+      when {
+        expression { params.VERACODE_SCA }
+      }
+      steps {
+        withCredentials([string(credentialsId: 'veracode-token', variable: 'SRCCLR_API_TOKEN'),
+                        nexusCredentials]) {
+          sh "bash .jenkins/scripts/veracode-sca.sh $SRCCLR_API_TOKEN"
+        }
+      }
+    }
+    stage('Package the app for Static testing') {
+      when {
+        expression { params.VERACODE_SAST }
+      }
+      steps {
+        withCredentials([veracodeCredentials, nexusCredentials]) {
+          script {
+            sh "bash .jenkins/scripts/package.sh ${SKIP_OPTS}"
+          }
         }
+      }
     }
-    post {
-        success {
-            slackSend(color: '#00FF00', message: "SUCCESSFUL: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' (${env.BUILD_URL})", channel: "${slackChannel}")
+    stage('Veracode SAST (Static testing)') {
+      when {
+        expression { params.VERACODE_SAST }
+      }
+      steps {
+        withCredentials([veracodeCredentials, nexusCredentials]) {
+          script {
+            veracode applicationName: "$VERACODE_SANDBOX",
+              teams: "Components",
+              canFailJob: true,
+              createProfile: true,
+              criticality: "High",
+              debug: true,
+              copyRemoteFiles: true,
+              fileNamePattern: '',
+              useProxy: false,
+              replacementPattern: '',
+              scanExcludesPattern: '',
+              scanIncludesPattern: '',
+              scanName: "${git_branch_name}-${currentBuild.number}-${currentBuild.startTimeInMillis}",
+              uploadExcludesPattern: '**/.cache/**/*.jar,**/*dep*/**/*.jar,**/lib/**/*.jar,**/*repo*/**/*.jar,**/it/**/*.jar,**/sample*.jar',
+              uploadIncludesPattern: '**/*.jar',
+              waitForScan: true,
+              vid: "$VERACODE_ID",
+              vkey: "$VERACODE_KEY"
+          }
         }
-        failure {
-            slackSend(color: '#FF0000', message: "FAILED: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' (${env.BUILD_URL})", channel: "${slackChannel}")
+      }
+    }
+  }
+  post {
+    always {
+      script {
+        if (params.JENKINS_DEBUG) {
+          jenkinsBreakpoint()
```

Comment:
don't get the goal of stopping here, any use-case?

Reply:
Jenkins is really complicated to debug when in kubernetes because the pod is immediately deleted.
I add this function to all my jenkins job, is is explained in the parameter description:
`booleanParam(
          name: 'DEBUG_BEFORE_EXITING',
          defaultValue: false,
          description: 'Add an extra step to the pipeline, allowing to keep the pod alive for debug purposes.')`

Reply:
I wonder how do you debug the jenkins pod? With witch tools?

Reply:
It is not technically "debugging" it allows accessing the pod and navigate inside the running system, allowing to check logs files and other files that are not keeps as artefact in the Jenkins job